src/launch/launch_config.cpp

```diff
@@ -127,6 +127,17 @@ bool ParseContext::shouldSkip(TiXmlElement* e)
 	return false;
 }
 
+
+bool ParseContext::checkEmptyString(std::string str)
+{
+	for(char& c : str) {
+		if(!isspace(c) or c!='\n')
```

Comment:
Please use `std::isspace`. Also, your condition does not make sense (every character except '\n' is detected here).

According to http://en.cppreference.com/w/cpp/string/byte/isspace '\n' should already be detected as space, so no need to handle it explicitly.

Reply:
Perhaps more elegantly: `return std::all_of(str.begin(), str.end(), std::isspace);`

Reply:
all_of is only available in C++11 and ROS uses by default C++03. Is it ok if I use this?

for(char& c : str)
{
	if(!isspace(c))
		return false;
}
return true;

Reply:
Erm, your range-based for loop is C++11 as well. rosmon uses C++11 everywhere, so don't hesitate to use the new features for cleaner code.

I just noticed that it should be
`return str.empty() || std::all_of(str.begin(), str.end(), std::isspace);`.

Reply:
Ah okay, I was not aware of the difficulties around `std::isspace`. The proposed solution looks good to me.